wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -964,12 +973,20 @@ public Address currentReceiveAddress() {
     @NonNull
     @Override
     public Flow<Coin> observeBalance(@NonNull Wallet.BalanceType balanceType) {
+        if (wallet == null) {
+            return FlowKt.emptyFlow();
```

Comment:
https://console.firebase.google.com/u/1/project/dash-wallet-firebase/crashlytics/app/android:hashengineering.darkcoin.wallet/issues/c46686534ce8391f9b22da2e3bb27a17?time=last-seven-days&types=crash&sessionEventKey=62B5F825012300013C1248FD38D1E101_1691303059962379795

What should we do in general if the wallet is null? I feel like there should be some kind of global handler for this case - recreate, reset the wallet, or something else.

Reply:
That is a good question.  There are some `wallet = null` cases in some other crashes with Krip Phones on Android 8.1.  And it is puzzling how the app gets in such a state, because it seems like the wallet must be loaded (and be non-null) before the code in this exception is executed.

A global handler is a good idea - we could log something to CrashLytics and tell the user to report the issue.  If we get the logs we might be able to tell what is happening.  Not sure at this point if the wallet needs to recreated, or reset or something else though.